### PR TITLE
Re-add app activate showing the window

### PR DESCRIPTION
**Pull request by stitesExpensify**
cc: @roryabraham since you reviewed the original PR (linked below) and I want to make sure I'm now accidentally breaking the auto-update functionality somehow

### Details
We removed this line [here](https://github.com/Expensify/Expensify.cash/pull/1318/files#diff-9276f078b750aa322d09bda48c79eb8034941df5b0f0f0c4f8b898cf9c652c41L141) which made it so that when you close the desktop app by clicking the `x` button or `cmd+w` you can no longer re-open the app.

### Fixed Issues
https://github.com/Expensify/Expensify/issues/156206

### Tests
1. Open the desktop app
2. Click the x in the top left
3. Make sure you can re-open the app from the task bar
4. Close the app with `cmd+w` 
5. Make sure you can re-open the app from the task bar

### Tested On
- [x] Desktop

### CODE REVIEW

**Review by roryabraham — CHANGES_REQUESTED** [review]
Hmmm I am suspicious this _might_ break the auto-update functionality, but I'm not sure. I know that the auto-update functionality basically works by relaunching the app with a special flag for the expected version. Then if the app version doesn't match the expected version, the browser window is hidden until the update is finished installing.

But the code in this PR tells the app window to display when the main process is activated.

I don't have a perfect understanding of the relationship between the `main` and `renderer` processes, but here's what I think could happen if we merged this PR:

1) An update becomes available, and is downloaded.
2) When the download is complete, the app relaunches w/ the new `--expected-update-version` flag.
3) The code from this PR would tell the window to show, but [this code here](https://github.com/Expensify/Expensify.cash/blob/ce5e480214bab5f67d9968c72b31bd0d02d49ea1/desktop/main.js#L190-L194) would tell it to hide. I'm not sure which would actually happen - and it might even be inconsistent due to async/race conditions.
4) But if the app window is shown when the update is in the middle of installing, then it would just be a blank white screen. Or maybe the old version would flash for a second then be hidden.

I'm really not sure, and maybe I'm just being paranoid because after all the update flow is a bit of an edge case. Not really sure the best way to handle this. Maybe just something like this would be safer:

```javascript
app.on('activate', () => {
    if (!expectedUpdateVersion || app.getVersion() === expectedUpdateVersion) {
        browserWindow.show();
    }
});
```

P.S: Sorry for the long-winded, scramble-brained review 😅

File: ubuntu-xenial-16.04-cloudimg-console.log [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[106438.241934] e1000 0000:00:08.0 enp0s8: Reset adapter
```
[the remark this answers is not in the document]
Comment (stitesExpensify) [reply ubuntu-xenial-16.04-cloudimg-console.log:1]: lol oops didn't even notice

**Comment by stitesExpensify** [comment]
@roryabraham updated to your code and the testing still works

**Review by roryabraham — APPROVED** [review]
Okay, I think this should be fine.

**Pull request merged**